tests/unit/suites/libraries/joomla/user/JAuthenticationTest.php

```diff
@@ -82,6 +82,7 @@ protected function tearDown()
 	{
 		// Reset the loaded plugins.
 		TestReflection::setValue('JPluginHelper', 'plugins', null);
+		$this->restoreFactoryState();
 
 		parent::tearDown();
 	}
```

Comment:
I'd have to look at PHPUnit_Framework_TestCase again but it has methods to backup the globals and validate our code didn't change them if we choose to call that.  I started using it in the Framework before we got everything in the web app mocked.

More useful for PHP globals than our app globals though.